src/main/groovy/org/shipkit/internal/gradle/e2e/E2ETestTask.java

```diff
@@ -0,0 +1,105 @@
+package org.shipkit.internal.gradle.e2e;
+
+import org.gradle.api.DefaultTask;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.shipkit.internal.gradle.git.CloneGitRepositoryTask;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.Arrays.asList;
+import static org.shipkit.internal.gradle.util.StringUtil.capitalize;
+
+/**
+ * Aggregates all e2e-related tasks. It can be configured to run e2e tests on provided repositories.
+ * Automatically added tasks clone client projects to '$buildDir/project-name-pristine' first, next clone project from 'pristine' to
+ * '$buildDir/project-name-work' and execute 'testRelease' task using the newest shipkit version
+ *
+ * Adds tasks:
+ * <ul>
+ *     <li>cloneProjectFromGitHub$projectName - {@link CloneGitRepositoryTask}</li>
+ *     <li>cloneProjectToWorkDir$projectName - {@link CloneGitRepositoryTask}</li>
+ *     <li>test$projectName - {@link RunTestReleaseTask}</li>
+ * </ul>
+ */
+public class E2ETestTask extends DefaultTask {
+
+    private static final Logger LOG = Logging.getLogger(E2ETestTask.class);
+
+    private List<String> repositories = new ArrayList<String>();
+
+    /**
+     * URL of repository which will be downloaded and e2e test will be run on it
+     */
+    public void addRepository(String repositoryUrl) {
```

Comment:
I like this!

It's nice because we can document the behavior in the task and have one task that runs all e2e tests.